src/components/NewExperiment.vue

```diff
@@ -1489,6 +1495,34 @@ export default Vue.extend({
     updateProgressValue() {
       this.submitProgressValue += this.itemWeight;
     },
+    isTableDisabled(key) {
+      if (key === "conditions") {
+        return false;
+      } else if (key === "samples") {
+        const mainField = this.tables.conditions.mainField;
+        if (
+          this.tables.conditions.items[0][mainField] &&
+          this.tables.conditions.isValid
+        ) {
+          return false;
+        }
+      } else {
+        const mainField = this.tables.samples.mainField;
+        if (
+          this.tables.samples.items[0][mainField] &&
+          this.tables.samples.isValid
+        ) {
+          return false;
+        }
+      }
+      return true;
+    },
+    getMessageforDisabledTable(key) {
+      if (key === "samples") {
+        return "You should enter valid condition first";
+      }
+      return "You should enter valid sample first";
```

Comment:
It is usually better to merge these messages with logic that determines isTableDisabled, e.g.
```ts
    isTableDisabled(key): false | string {
      if (key === "conditions") {
        return false;
      } else if (key === "samples") {
        const mainField = this.tables.conditions.mainField;
        if (
          this.tables.conditions.items[0][mainField] &&
          this.tables.conditions.isValid
        ) {
          return false;
        } else {
          return "You should enter valid condition first";
        }
      } else {
        const mainField = this.tables.samples.mainField;
        if (
          this.tables.samples.items[0][mainField] &&
          this.tables.samples.isValid
        ) {
          return false;
        } else {
          return "You should enter valid sample first";
        }
      }
    },
```

or, in more complex situations, `return { disabled: true, msg: "because.." }`

Reply:
Perfect, thanks!